crates/bevy_solari/src/pathtracer/mod.rs

```diff
@@ -65,3 +58,15 @@ impl Plugin for PathtracingPlugin {
 pub struct Pathtracer {
     pub reset: bool,
 }
+
+// We only want to add these render graph nodes and edges if Solari required features are present.
+// Making this a system that runs at RenderStartup allows a run condition to check for required
+// features first.
+fn add_solari_pathtracing_render_graph_nodes(world: &mut World) {
+    world
```

Comment:
I feel like we should support the slightly nicer .add_render_graph_node directly on the RenderGraph and then query that. It's a bit weird to get the entire world just for that. It would be more explicit that way.

Reply:
Not to do in this PR to be clear, but as a follow up or before this PR merges it would be worth looking into it.

Reply:
Good point! I was definitely more focused on keeping the diff small. One advantage of this heavy-handed approach is we have to do significantly less error handling - all the error handling and logging is already in add_render_graph_node.

I'd be down to do that as a followup though.

Reply:
fallible systems? iirc theres a When or maybe its If now. `render_graph: If<RenderGraph>` etc

Reply:
It turns out that's not enough: We need &mut World anyway because to init a ViewNodeRunner you need to use FromWorld. So either way we'd need exclusive world access here.